src/IdentityServer/Services/Default/NumericUserCodeGenerator.cs

```diff
@@ -36,27 +36,7 @@ public class NumericUserCodeGenerator : IUserCodeGenerator
     /// <returns></returns>
     public Task<string> GenerateAsync()
     {
-        var next = Next(100000000, 999999999);
+        var next = RandomNumberGenerator.GetInt32(100000000, 1000000000);
```

Comment:
The one built in to .NET has an exclusive upper bound, while the previous one was inclusive, hence the upper bound of `1_000_000_000`.